azure-iot-device/tests/common/pipeline/test_pipeline_ops_base.py

```diff
@@ -7,64 +7,268 @@
 import pytest
 import logging
 from azure.iot.device.common.pipeline import pipeline_ops_base
-from tests.common.pipeline import pipeline_data_object_test
+from tests.common.pipeline import pipeline_ops_test
 
 this_module = sys.modules[__name__]
 logging.basicConfig(level=logging.DEBUG)
+pytestmark = pytest.mark.usefixtures("fake_pipeline_thread")
 
 
-@pytest.mark.describe("PipelineOperation")
-class TestPipelineOperation(object):
-    @pytest.mark.it("Can't be instantiated")
-    def test_instantiate(self, mocker):
-        with pytest.raises(TypeError):
-            pipeline_ops_base.PipelineOperation(mocker.MagicMock())
+class ConnectOperationTestConfig(object):
+    @pytest.fixture
+    def cls_type(self):
+        return pipeline_ops_base.ConnectOperation
 
+    @pytest.fixture
+    def init_kwargs(self, mocker):
+        kwargs = {"callback": mocker.MagicMock()}
+        return kwargs
 
-pipeline_data_object_test.add_operation_test(
-    cls=pipeline_ops_base.ConnectOperation, module=this_module
+
+pipeline_ops_test.add_operation_tests(
+    test_module=this_module,
+    op_class_under_test=pipeline_ops_base.ConnectOperation,
+    op_test_config_class=ConnectOperationTestConfig,
 )
-pipeline_data_object_test.add_operation_test(
-    cls=pipeline_ops_base.DisconnectOperation, module=this_module
+
+
+class DisconnectOperationTestConfig(object):
+    @pytest.fixture
+    def cls_type(self):
+        return pipeline_ops_base.DisconnectOperation
+
+    @pytest.fixture
+    def init_kwargs(self, mocker):
+        kwargs = {"callback": mocker.MagicMock()}
+        return kwargs
+
+
+pipeline_ops_test.add_operation_tests(
+    test_module=this_module,
+    op_class_under_test=pipeline_ops_base.DisconnectOperation,
+    op_test_config_class=DisconnectOperationTestConfig,
 )
-pipeline_data_object_test.add_operation_test(
-    cls=pipeline_ops_base.ReconnectOperation, module=this_module
+
+
+class ReconnectOperationTestConfig(object):
+    @pytest.fixture
+    def cls_type(self):
+        return pipeline_ops_base.ReconnectOperation
+
+    @pytest.fixture
+    def init_kwargs(self, mocker):
+        kwargs = {"callback": mocker.MagicMock()}
+        return kwargs
+
+
+pipeline_ops_test.add_operation_tests(
+    test_module=this_module,
+    op_class_under_test=pipeline_ops_base.ReconnectOperation,
+    op_test_config_class=ReconnectOperationTestConfig,
 )
-pipeline_data_object_test.add_operation_test(
-    cls=pipeline_ops_base.EnableFeatureOperation,
-    module=this_module,
-    positional_arguments=["feature_name", "callback"],
+
+
+class EnableFeatureOperationTestConfig(object):
+    @pytest.fixture
+    def cls_type(self):
+        return pipeline_ops_base.EnableFeatureOperation
+
+    @pytest.fixture
+    def init_kwargs(self, mocker):
+        kwargs = {"feature_name": "some_feature", "callback": mocker.MagicMock()}
```

Comment:
>kwargs = {"feature_name": "some_feature" [](start = 8, length = 40)

Maybe I missed it, but is there a test that verifies that op.feature_name gets set to some_feature? (same for all other op attributes)  There used to be.  If you removed this, I want to make sure it was intentional.